src/video_utils.py

```diff
@@ -201,9 +265,21 @@ def download_video(url):
         for filename in os.listdir(temp_dir):
             if filename.endswith(".mp4"):
                 result_path = os.path.join(temp_dir, filename)
-                break  # Exit the loop once the file is found
+                return result_path  # Exit the loop once the file is found
+
     except subprocess.CalledProcessError as e:
         error("Error downloading video: %s", e)
+        if "[Instagram]" in str(e) and "No video formats found!" in str(e):
+            debug("yt-dlp failed for Instagram URL, trying gallery-dl")
+        try:
+            result_path = download_instagram_images(url, temp_dir)
```

Comment:
downloading `images` inside `download_video` function

Reply:
> downloading `images` inside `download_video` function

I think it worth to rename some functions here. The main idea that if content is video, that original code will handle it, and in case if fail specifically for Instagram, we have fallback to `gallery-dl`. And for sure, that ranaiming should be here, as current implementation use `download_instagram_images` with `gallery-dl` for case of mixed content. I tested with link that return 51 media, some of them is video, and some of them is picture.

Reply:
renamed